src/worker/pbs_info.rs

```diff
@@ -0,0 +1,52 @@
+use std::process::Command;
+use std::{str, env};
+use std::time::Duration;
+use serde_json::Value;
+
+type TimeNumberFormat = u64;
+
+
+fn process_time_format(raw_time: String) -> TimeNumberFormat {
+    let numbers_str: Vec<&str> = raw_time.split(":").collect();
+    let numbers: Vec<TimeNumberFormat> = numbers_str.iter()
+        .map(|x| x.parse::<TimeNumberFormat>().unwrap())
+        .collect();
+    let seconds = numbers[0] * 3600 + numbers[1] * 60 + numbers[2];
+    seconds
+}
+
+
+fn get_time(job_id: &str, data: String) -> serde_json::Result<Duration> {
+    let data_json: Value = serde_json::from_str(data.as_str())?;
+
+    let walltime = process_time_format(String::from(data_json["Jobs"][job_id]["Resource_List"]["walltime"].as_str().unwrap()));
+    let used = process_time_format(String::from(data_json["Jobs"][job_id]["resources_used"]["walltime"].as_str().unwrap()));
+
+    Ok(Duration::from_secs(walltime - used))
+}
+
+
+fn get_remaining_time() -> serde_json::Result<Duration> {
```

Comment:
You can return `anyhow::Result` so that you can replace the panics with `?`. For now it's probably enough.